From ad997620377a29eecd640caf00b4e7aecabfc5bf Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 08 Nov 2021 18:13:29 +0200
Subject: [PATCH] [SW-0] gaudi2: pll: add post divide power down bit

The "Post divide power down" bit was not used as a part of the
configuration sequence.
This resulted in ARM being stuck as soon as the first IF PLL was
configured.
Add the FOUTPOSTDIVEN bit to the reset sequence.

Change-Id: I4464cc19d0e249bdaf3e20a31b0f63d1aed8e341
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/pll.c b/arch/gaudi2/common/pll.c
index 172bec2..77ad853 100644
--- a/arch/gaudi2/common/pll.c
+++ b/arch/gaudi2/common/pll.c
@@ -51,7 +51,7 @@
 }
 
 #define GAUDI2_PLL_CFG			0x100
-#define GAUDI2_PLL_CFG_EN		BIT(0)
+#define GAUDI2_PLL_CFG_EN		(BIT(8) | BIT(0))
 #define GAUDI2_PLL_CFG_LOCK		BIT(16)
 #define GAUDI2_PLL_DIV_SEL_DELAY	0x1E0
 
@@ -81,7 +81,7 @@
 {
 	u32 reg = 0;
 
-	/* put the PLL into reset by writing 0 to CTRL_CFG.EN */
+	/* put the PLL into reset by writing 0 to CTRL_CFG.EN and CTRL_CFG.FOUTPOSTDIVEN */
 	hl_pll_reset(cfg, false);
 
 	/* FBDIV - bit 0
@@ -98,7 +98,7 @@
 	/* after updating the coefficients we need to wait >=1us */
 	udelay(GAUDI2_PLL_MAX_WAIT_TIME);
 
-	/* enable the PLL by writing a value of 1 to CTRL_CFG.EN*/
+	/* enable the PLL by writing a value of 1 to CTRL_CFG.EN and CTRL_CFG.FOUTPOSTDIVEN */
 	hl_pll_reset(cfg, true);
 }
 
